src/tiler.coffee

```diff
@@ -21,130 +21,152 @@
+      return $enteringTile
 
     #
     # PRIVATE METHODS
     #
-    _getAnimationClass: ->
-      @$enterTile.data('tiler-animation') || @element.data('tiler-animation') || ''
 
-    _transitionCss: (animationClass, animation) ->
-      animationClass = animation if typeof animation == 'string'
+    # Find tile with various values
+    #
+    _getTile: (tileValue) ->
+      $tile = switch typeof tileValue
+        # jquery object or dom element
+        when 'number'
+          @$tiles.eq tileValue - 1
+        when 'string'
+          $("##{tileValue}", @element)
+        else
+          $(tileValue, @element)
 
-      position = if animation == false
-        'end'
+      # if no tile is found, return the first tile
+      if $tile.length
+        $tile
       else
-        'start'
+        @$tiles.eq 0
 
-      enterTileIndex = @$tiles.index @$enterTile
+    # get HUMAN COUNTABLE index (eg starting at 1, not 0)
+    #
+    _getTileIndex: ($tile) ->
+      @$tiles.index($tile) + 1
+
+    _getAnimationClass: ($tile) ->
+      $tile.data('tiler-animation') || @element.data('tiler-animation') || ''
+
+    _transitionCss: ($enteringTile, $exitingTile, animation) ->
+      # Get animation class
+      animationClass = if typeof animation == 'string'
+        animation
+      else
+        @_getAnimationClass $enteringTile
 
-      # Add reverse class if supported and navigating in reverse order (according to the dom)
-      reverseClass = if @options.isReversible && !@_isNavigatingForward(enterTileIndex)
+      # Get reverse class
+      reverseClass = if @options.isReversible && @_getTileIndex($enteringTile) < @_getTileIndex(@$currentActiveTile)
         'reverse'
       else
         ''
 
+      # Get position class
+      positionClass = if animation == false
+        'end'
+      else
+        'start'
+
       # Disable animations
       @element.addClass 'animation-disabled'
 
       # Build tile starting position animations classes
-      enterStartClass = "tiler-tile #{animationClass} active enter #{reverseClass} #{position}"
-      exitStartClass = "tiler-tile #{animationClass} previous exit #{reverseClass} #{position}"
-      otherTileClass = 'tiler-tile'
-
-      # Set tile classes
-      @$enterTile.attr 'class', enterStartClass
-      @$exitTile.attr 'class', exitStartClass
-      @$tiles.not(@$enterTile).not(@$exitTile).attr 'class', otherTileClass
+      $enteringTile.attr 'class', "tiler-tile #{animationClass} active enter #{reverseClass} #{positionClass}"
+      $exitingTile.attr 'class', "tiler-tile #{animationClass} previous exit #{reverseClass} #{positionClass}"
+      @$tiles.not($enteringTile).not($exitingTile).attr 'class', 'tiler-tile'
 
       # setTimeout needed to give the browser time to repaint the tiles (if neccessary) with the animation starting position
-      unless animation == false
+      if animation == false
+        @_finalizeNewTiles $enteringTile, $exitingTile
+      else
         setTimeout =>
```

Comment:
i guess since it isn't directly triggering a repaint then it makes sense. why the setTimeout in the first place?